helpers/recordSync.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2015  TF2Stadium
+// Use of this source code is governed by the GPLv3
+// that can be found in the COPYING file.
+
+package helpers
+
+import (
+	"reflect"
+	"sync"
+)
+
+type record struct {
+	id      uint
+	recType reflect.Type
+}
+
+var mutexStore = make(map[record]*sync.Mutex)
+
+func LockRecord(id uint, recType interface{}) {
```

Comment:
Go maps aren't thread safe so access to mutexStore itself should be synchronized. Also, this will leak memory because mutexes are never removed from the map. For now, just put a TODO comment, later I think we'll need to write a goroutine that will periodically clean up this map and some other places in our code.